docs/network-binding-plugin.md

```diff
@@ -357,3 +357,111 @@ The expected artifacts include:
 > **Note**: In order to consume a sidecar, the image needs to be available in
 > an accessible registry.
 > Cluster admins may need to make the sidecar image available on their registry.
+
+## The Complete Plugin
+
+We have covered so far network binding plugins that utilize core domain attachment
+types and/or custom sidecars. While many network bindings can find these
+integration points sufficient, there is a missing piece.
+
+Network bindings that require pod network changes, should use a CNI plugin.
+
+### Using a custom CNI plugin
+
+A CNI plugin is a program that is called by the runtime when the pod is created.
+The program is expected to interpret a standard CNI format input and perform
+changes in the pod network namespace.
+
+> **Note**: The CNI plugin discussed in this section is used for implementing
+> a network binding plugin. It is not to be confused with the CNI plugin that
+> connects the pod to a network on the node.
+
+#### Custom CNI plugin requirements
+
+In order to call multiple times CNI plugins, the cluster requires
+Multus to be deployed.
+
+In addition, the CNI plugin itself needs to be deployed on the cluster nodes.
+
+> **Note**: It is out of this document scope to elaborate on how a binary can
+> be deployed on cluster nodes and kept up-to-date.
+
+A network binding CNI plugin belongs to one specific binding, and it is
```

Comment:
It is not accurate, different plugins may use the same plugin.

Reply:
I think the common case is that we will have a network binding that may have a CNI plugin. That CNI plugin will be tailored to the need of the network binding.
While the scenario that we will use the same custom binding CNI plugin for two different bindings, I think it will be an exception.

If I try to add this exception in the explanation, it gets messy.
I am not sure how to phrase it better and not confuse the reader.